tests/src/Traits/CommandTesterTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Robo\Traits;
+
+use Consolidation\AnnotatedCommand\CommandFileDiscovery;
+use Robo\Robo;
+use Robo\Runner;
+use Symfony\Component\Console\Tester\CommandTester;
+
+trait CommandTesterTrait
+{
+    /** @var string */
+    protected $appName;
+
+    /** @var string */
+    protected $appVersion;
+
+    /** @var string|array|null */
+    protected $commandClasses;
+
+    /** @var Runner */
+    protected $runner;
+
+    /**
+     * Setup the tester.
+     *
+     * @param string|array|null $commandClasses
+     */
+    public function setupCommandTester($commandClasses = null)
+    {
+        // Define our invariants for our test.
+        $this->runner = new Runner();
+        if (!is_null($commandClasses)) {
+            $this->commandClasses = $commandClasses;
+        }
+    }
+
+    /**
+     * @param $command_string
+     * @param array $inputs
+     * @param array $command_extra
+     * @param string|array|null $commandClasses
+     * @return array
+     */
+    protected function executeCommand($command_string, $inputs = [], $command_extra = [], $commandClasses = null)
+    {
+        if (is_null($commandClasses)) {
+            $commandClasses = $this->commandClasses;
+        }
+        $app = $this->runner->getAppForTesting($this->appName, $this->appVersion, $commandClasses);
```

Comment:
I think you could get rid of the if above and use `$commandClasses ?? $this->commandClasses` here. I think ?? was added in 7.0.  We only need 7.1+ here.